APIs/schemas/network-link.json

```diff
@@ -1,45 +1,59 @@
-  "description": "Describes a network link. A network link consists of two peers. A peer is either a network device or an endpoint. But a link cannot have two endpoints as peers. Links are always bidirectional. That is, the order of the peers in this entity carries no special meaning. When a link is between a network device (switch) and an endpoint, the port_id for the endpoint peer must be null.",
+  "description": "Describes a network link. A network link consists of two peers. A peer is either a network device or an endpoint. But a link cannot have two endpoints as peers. Links are always bidirectional. That is, the order of the peers in this entity carries no special meaning.",
   "required": [
+    "id",
     "peers",
     "speed"
   ],
   "properties": {
+    "id": {
+      "description": "Globally unique identifier for the network link",
+      "type": "string",
+      "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+    },
     "peers": {
       "description": "List of peer devices in a link. Peer can be a network device (switch) or an endpoint. There must be exactly two peers in the list.",
       "type": "array",
       "items": {
-        "type": "object",
-        "required": [
-          "device_id",
-          "port_id"
-        ],
-        "properties": {
-          "device_id": {
-            "description": "Globally unique identifier of a network device or an endpoint.",
-            "type": "string",
-            "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
-          },
-          "port_id": {
-            "description": "The ifAlias of the interface of the network device or endpoint.",
-            "anyOf": [
-              {
+        "oneOf": [
+          {
+            "type": "object",
+            "required": [
+              "device_id",
+              "port_id"
+            ],
+            "properties": {
+              "device_id": {
+                "description": "Globally unique identifier of a network device.",
                 "type": "string",
-                "description": "The ifAlias of the interface of the peer device. This option MUST not be used if the peer is an endpoint."
+                "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
               },
-              {
-                "type": "null",
-                "description": "When the peer is an endpoint (instead of network device) the port_id MUST be null. A null value will explicitly identify this peer as an endpoint. An endpoint has only one port, so a port_id value is redundant."
+              "port_id": {
+                "description": "The ifAlias of the interface of the network device.",
+                "type": "string"
+              }
+            }
+          },
+          {
+            "type": "object",
+            "required": [
+              "endpoint_id"
+            ],
+            "properties": {
```

Comment:
I can never quite decide whether adding `"additionalProperties": false` is a good thing or not, when considering future extensibility. With versioned APIs, it could be the right thing to do?